access/types/forms.py

```diff
@@ -499,27 +510,40 @@ def row_options(self, configuration, row):
     def grade_checkbox(self, configuration, value, hints=None):
         hints = hints or []
         correct_count = 0
-
-        # All correct required if any configured
+        wrong_answers = 0
+        non_neutral_count = 0
+        # All correct required if any configured and partial-points not set
         correct = True
         i = 0
         for opt in configuration.get("options", []):
             name = self.option_name(i, opt)
-            if opt.get("correct", False):
+            correct_answer = opt.get("correct", False)
+            # correct_answer may be boolean or string "neutral"
+            if correct_answer is True:
                 correct_count += 1
+                non_neutral_count += 1
                 if name not in value:
+                    wrong_answers += 1
                     correct = False
                     self.append_hint(hints, opt)
-            elif not value is None and name in value:
-                correct = False
-                self.append_hint(hints, opt)
+            elif correct_answer is False:
+                non_neutral_count += 1
+                if not value is None and name in value:
+                    correct = False
+                    wrong_answers += 1
+                    self.append_hint(hints, opt)
             i += 1
 
+        #if partial_points are set, correct is scaled to 0-1 float, to be factored to max points
+        #The zero points limit is in the half of the correct answers
+        if configuration.get("partial_points"):
+            correct = float(non_neutral_count / 2 - wrong_answers) / (non_neutral_count / 2)
```

Comment:
If there is only one choice (checkbox), is this going to divide by zero?

Reply:
Yes. There is no sense to use "partial_points" with just one non_neutral. This situation can occur in randomized checkbox questions so this should still be fixed.

Reply:
Rare occasion but still possible, especially with randomized checkbox. Can be avoided with converting 2 to float 2.0.

Reply:
Now that you use floats, if the student answers correctly and `wrong_answers` is zero, is `correct` really guaranteed to equal `1` since floating point arithmetic is inherently inaccurate?